src/pages/collab/index.tsx

```diff
@@ -0,0 +1,287 @@
+import { useEffect, useRef, useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faXmark } from "@fortawesome/free-solid-svg-icons";
+
+import { api } from "~/utils/api";
+
+import LoadingIcon from "~/components/LoadingIcon";
+
+const MatchRequestPage = () => {
+  const [pageState, setPageState] = useState({
+    difficulty: -1,
+    category: "",
+    id: "",
+    response: "",
+    statusMessage: "Searching for partner...",
+    isWaiting: false,
+    requestFailed: false,
+    hasSetDifficulty: false,
+    hasSetCategory: false,
+    isDifficultyMissing: true,
```

Comment:
What's the difference between hasSetCategory and isCategoryMissing? Aren't the two just not of each other?